### PR TITLE
Grab the values for share propagation manually from the DB

### PR DESCRIPTION
**Pull request by nickvergessen**
@icewind1991 should work like this? or did I miss another spot where the values are required?

cc @PVince81 @MorrisJobke 

Fix https://github.com/owncloud/enterprise/issues/1149

### CODE REVIEW

File: apps/files_sharing/lib/propagation/recipientpropagator.php [hunk 1]
```diff
@@ -119,6 +126,32 @@ protected function getDirtyShares($shares) {
...
+	protected function getPropagationTimestampForShares(array $sharePropagations) {
+		$sql = \OC::$server->getDatabaseConnection()->getQueryBuilder();
+		$shareIds = array_keys($sharePropagations);
+
+		$sql->select(['configkey', 'configvalue'])
+			->from('appconfig')
+			->where($sql->expr()->eq('appid', $sql->createParameter('appid')))
+			->andWhere($sql->expr()->in('configkey', $sql->createParameter('shareids')))
+			->setParameter('appid', 'files_sharing', \PDO::PARAM_STR)
+			->setParameter('shareids', $shareIds, Connection::PARAM_INT_ARRAY);
```
Comment (rullzer) [inline apps/files_sharing/lib/propagation/recipientpropagator.php]: Just to be safe we should paginate here. Some db's result to seq scan on more than `N` entries... (where I can't remember the exact value of `N` right now)

**Comment by nickvergessen** [comment]
I ran the smashbox propagation tests manually
* smashbox/lib/owncloud/test_sharePropagation.py :white_check_mark: 
* smashbox/lib/owncloud/test_sharePropagationGroups.py :white_check_mark: 
* smashbox/lib/owncloud/test_sharePropagationInside.py :x: 
* smashbox/lib/owncloud/test_sharePropagationInsideGroups.py :white_check_mark: 

which is the same pattern as with stable8.2

File: apps/files_sharing/lib/propagation/recipientpropagator.php (hunk 1 above)
Comment (nickvergessen) [inline apps/files_sharing/lib/propagation/recipientpropagator.php]: sql 50 iirc

**Comment by rullzer** [comment]
Etags are still changing for me... (basically I did a simplified version of the `test_sharePropagation.py` by hand).

:+1:

**Comment by icewind1991** [comment]
Code looks good :+1:

**Comment by MorrisJobke** [comment]
I tested this in different scenarios and it works the same way as the stable8.2 version worked :+1:

**Comment by MorrisJobke** [comment]
@karlitschek For this backport

**Comment by PVince81** [comment]
@karlitschek to clarify, this is a fix that goes *only* to stable8.2 since 9.0 uses a different approach.

**Comment by karlitschek** [comment]
backport is fine if we have extensive testing. @pvince81 your call if the current tests are enoug if if we should do more.

File: lib/private/appconfig.php [hunk 2]
```diff
@@ -273,6 +274,17 @@ protected function loadConfigValues() {
...
+		// we are not loading the propagation timestamps by default anymore.
+		// The code relying on those values has been adjusted to grab the values
+		// manually. In 9.0 the propagation was changed to not be stored in this
+		// table anymore.
+		$sql->where($sql->expr()->orX(
+				$sql->expr()->neq('appid', $sql->createParameter('appid')),
+				$sql->expr()->in('configkey', $sql->createParameter('legit_configs'))
+			))
+			->setParameter('appid', 'files_sharing', \PDO::PARAM_STR)
+			->setParameter('legit_configs', ['enabled', 'installed_version', 'types'], Connection::PARAM_STR_ARRAY);
```
Comment (PVince81) [inline lib/private/appconfig.php]: Do we need "ocsid" ?

Does it mean that every config value that isn't preloaded will still properly be retrieved through other methods ?

Comment (PVince81) [inline lib/private/appconfig.php]: Never mind, I understand the query now.

**Comment by PVince81** [comment]
This looks fine :+1: 
We have smashbox tests that represent the many combinations of cases that I used to do manually in the past.

**Comment by scolebrook** [comment]
This change breaks the 2 checkboxes that enable or disable federation in the admin page for 8.2.4 and above. For those two settings, incoming_server2server_share_enabled and outgoing_server2server_share_enabled, the database says 'no' but the functions in /apps/files_sharing/lib/helper.php always get the default value of 'yes' back from getValue in appconfig.

This also results in the federation section being displayed in the personal settings page displaying that lovely long objectguid@server.hostname federation address.

I'm changing the defaults in helper.php to no to work around this issue until a proper fix can be found. We're upgrading from 8.2.3 to 8.2.5 today due to the security fix and I don't have time to wait for a formal fix.

**Comment by gig13** [comment]
@PVince81 @bboule Any way we can disable the Federation? Even a workaround to the theme?

**Comment by scolebrook** [comment]
The following is needed for the legit_configs parameter:
```php
			->setParameter('legit_configs', ['enabled', 'installed_version', 'types',
				'incoming_server2server_share_enabled',
				'outgoing_server2server_share_enabled'], Connection::PARAM_STR_ARRAY);
```
This allows the server2server configs to be read successfully. I've tested this on 8.2.5 and it works.

**Comment by nickvergessen** [comment]
I will have a look tomorrow

**Comment by nickvergessen** [comment]
Patch in https://github.com/owncloud/core/pull/24656

**Comment by PVince81** [comment]
Ah yes, this regression was reported here https://github.com/owncloud/core/issues/24547

@scolebrook thanks for finding the breaking PR and thanks @nickvergessen for the patch..

**Pull request merged**